### PR TITLE
added patch endpoint for /api/articles/:article_id

### PR DESCRIPTION
**Pull request by smlbrd**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -113,6 +113,87 @@ describe('GET /api/articles/:article_id', () => {
...
+describe('PATCH /api/articles/:article_id', () => {
+  test('201: Responds with article including updated vote count for given article_id', () => {
+    const testBody = {
+      inc_votes: 1,
+    };
+
+    return request(app)
+      .patch('/api/articles/2')
+      .send(testBody)
+      .expect(201)
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:125]: A `200` is fine here - a `201` is the status "created" but a patch is an "update" in the world of CRUD.

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -20,3 +22,17 @@ exports.getArticleById = (req, res, next) => {
...
+
+exports.patchArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+
+  checkArticleExists(article_id)
+    .then(() => {
+      return updateArticleById(article_id, inc_votes)
+        .then((article) => {
+        res.status(201).send({ article });
```
Comment (nc-mezz-davies) [inline controllers/articles.controllers.js:34]: As per above comment. Remember to alter the test first, see it fail then amend to pass.

File: endpoints.json [hunk 3]
```diff
@@ -80,5 +80,22 @@
...
+      "author": "icellusedkars",
+      "body": "Call me Mitchell. Some years ago—never mind how long precisely—having little or no money in my purse, and nothing particular to interest me on shore, I thought I would buy a laptop about a little and see the codey part of the world. It is a way I have of driving off the spleen and regulating the circulation. Whenever I find myself growing grim about the mouth; whenever it is a damp, drizzly November in my soul; whenever I find myself involuntarily pausing before coffin warehouses, and bringing up the rear of every funeral I meet; and especially whenever my hypos get such an upper hand of me, that it requires a strong moral principle to prevent me from deliberately stepping into the street, and methodically knocking people’s hats off—then, I account it high time to get to coding as soon as I can. This is my substitute for pistol and ball. With a philosophical flourish Cato throws himself upon his sword; I quietly take to the laptop. There is nothing surprising in this. If they but knew it, almost all men in their degree, some time or other, cherish very nearly the same feelings towards the the Vaio with me.",
```
Comment (nc-mezz-davies) [inline endpoints.json:95]: You could assert the data type of string to tidy up the test. So long as you assert the id and updated votes explicitly that is fine

**Review by nc-mezz-davies — COMMENTED** [review]
Just a few small comments but looking good other than that 🙂

File: __tests__/app.test.js (hunk 1 above)
Comment (smlbrd) [reply __tests__/app.test.js:125]: I've updated the status code to 200 :saluting_face:

File: controllers/articles.controllers.js (hunk 2 above)
Comment (smlbrd) [reply controllers/articles.controllers.js:34]: As above, updated test to 200, updated code to pass failing test :raised_hands:

File: endpoints.json (hunk 3 above)
Comment (smlbrd) [reply endpoints.json:95]: I've done another pass over these tests to make explicit assertions about the article_id I'm expecting - thanks Mezz! :saluting_face:

**Pull request merged**